lib/Scheduler/bb_spill.cpp

```diff
@@ -391,6 +418,11 @@ InstCount BBWithSpill::CmputCost_(InstSchedule *sched, COST_COMP_MODE compMode,
   sched->SetSpillCosts(spillCosts_);
   sched->SetPeakRegPressures(peakRegPressures_);
   sched->SetSpillCost(crntSpillCost_);
+
+  for (auto it = recordedCostFunctions.begin();
+       it != recordedCostFunctions.end(); ++it)
+    storeExtraCost(sched, *it);
```

Comment:
```
for (const auto& CostFunction : recordedCostFunctions)
  storeExtraCost(sched, CostFunction);
```

Reply:
Thanks for mentioning this.  I will implement this suggestion

Reply:
My one change to your suggested code is that I am not using the const reference to the cost function.  `recordedCostFunctions` stores 32 bit integers (SPILL_COST_FUNCTION is an enum).  Copying a 32 bit integer is no more costly than de-referencing the refrence/pointer to an int(since const refrences to integers are usually converted to int copies by the compiler anyway).
```
  for (SPILL_COST_FUNCTION CostFunction : recordedCostFunctions)
    storeExtraCost(sched, CostFunction);
```